### PR TITLE
Allow download-frozen-image.sh to work on user images too

### PR DESCRIPTION
**Pull request by tianon**
To account for "/" not working in filenames, we replace it with "_" for our temporary files (that exist only to emulate Bash 4's associative arrays in Bash 3).

cc @jfrazelle

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

File: contrib/download-frozen-image.sh [hunk 1]
```diff
@@ -42,6 +42,8 @@ while [ $# -gt 0 ]; do
 	[ "$tag" != "$imageTag" ] || tag='latest'
 	tag="${tag%@*}"
 
+	imageFile="${image////_}" # "/" can't be in filenames :)
```
Comment (jlhawn) [inline contrib/download-frozen-image.sh]: maybe I'm just crazy, but it seems easier to parse as `${image//\//_}`

Comment (tianon) [inline contrib/download-frozen-image.sh]: That sounds sane, but I think we need to test that under Bash 3 in Windows before we say for sure: (this is my testing on Bash 4 in Linux)
```console
$ a='a/b/c'
$ echo "${a//\//_}"
a_b_c
```

Comment (thaJeztah) [inline contrib/download-frozen-image.sh]: FYI, OS X is still on bash 3 too; https://github.com/docker/compose/issues/1386#issuecomment-99198672

Comment (tianon) [inline contrib/download-frozen-image.sh]: Jeez, even Debian Squeeze (now `oldoldstable`) and Ubuntu Lucid (10.04) have Bash 4.0. :jack_o_lantern:

Comment (thaJeztah) [inline contrib/download-frozen-image.sh]: @tianon iirc it had to do with Bash changing to another license at some point, which wasn't compatible with the OS X Licensing? I know it s#cks

Comment (thaJeztah) [inline contrib/download-frozen-image.sh]: If it really becomes troublesome we can advise to install a more recent version via Homebrew or something, of course (LOL, or follow Jess's tutorial on installing Linux)

Comment (tianon) [inline contrib/download-frozen-image.sh]: Score!
```console
root@4b34ca1e225f /git
$ bash --version
GNU bash, version 3.1.23(6)-release (i686-pc-msys)
Copyright (C) 2005 Free Software Foundation, Inc. 

root@4b34ca1e225f /git
$ a='a/b/c'

root@4b34ca1e225f /git
$ echo "${a//\//_}"
a_b_c
```

Updating now!

Comment (tianon) [inline contrib/download-frozen-image.sh]: Updated!

File: contrib/download-frozen-image.sh [hunk 2]
```diff
@@ -90,10 +92,12 @@ done
 echo -n '{' > "$dir/repositories"
 firstImage=1
 for image in "${images[@]}"; do
+	imageFile="${image////_}" # "/" can't be in filenames :)
```
Comment (tianon) [inline contrib/download-frozen-image.sh]: Doh, missed one.

Comment (tianon) [inline contrib/download-frozen-image.sh]: Updated again!

File: contrib/download-frozen-image.sh (hunk 1 above)
Comment (thaJeztah) [inline contrib/download-frozen-image.sh]: 👯✌️

**Pull request merged**